Engines/Wine/Engine/Versions/script.js

```diff
@@ -1,8 +1,139 @@
-/* exported LATEST_STABLE_VERSION */
-module.LATEST_STABLE_VERSION = "4.0.3";
-/* exported LATEST_DEVELOPMENT_VERSION */
-module.LATEST_DEVELOPMENT_VERSION = "4.21";
-/* exported LATEST_STAGING_VERSION */
-module.LATEST_STAGING_VERSION = "4.21";
-/* exported LATEST_DOS_SUPPORT_VERSION */
-module.LATEST_DOS_SUPPORT_VERSION = "4.0";
+const {cat, touch} = include("utils.functions.filesystem.files");
+const Downloader = include("utils.functions.net.download");
+const propertyReader = Bean("propertyReader");
+
+/**
+ * Sorts an array of Wine versions in place
+ * @param {array} versions The versions array (see packages in versions json)
+ * @returns {void}
+ */
+function sortVersions(versions) {
+    versions.sort((a, b) =>
+    {
+        // check version format
+        const versionRegExp = /^(\d+\.\d+(\.\d+)?)(.*)?$/;
+        if (!versionRegExp.test(a.version)) {
+            throw new Error(`invalid Wine version "${a.version}`);
+        }
+        if (!versionRegExp.test(b.version)) {
+            throw new Error(`invalid Wine version "${a.version}`);
+        }
+
+        // sort
+        const aVersionParts = a.version.match(versionRegExp);
+        const bVersionParts = b.version.match(versionRegExp);
+
+        const aVersionNumbers = Array.from(aVersionParts[1].split('.')).map(item => Number(item));
+        const bVersionNumbers = Array.from(bVersionParts[1].split('.')).map(item => Number(item));
+
+        // ensure that cases where not all version numbers (major, minor, patch) are set are handled correctly
+        const maxVersionIdx = 2;
+        let i;
+        for (i = 0; i <= maxVersionIdx; i++) {
+            if (typeof aVersionNumbers[i] === 'undefined') {
```

Comment:
or `if (!aVersionNumbers[i])` ?

Reply:
Do we want that? It would enter the code if `aVersionNumbers[i]` is 0.

Reply:
You are right.

Just asking but does `aVersionNumbers[i] == undefined` work?

Reply:
I've checked the sorted list and it looked ok.

Reply:
Just realized that you are referring to `==` vs `===`. I have no clue. I would assume it should work as well.

Reply:
I'm not sure if `===` works as well therefore I wanted to be sure and wrote `==` :D

Reply:
Well, the current implementation works. Everything else I can't say with certainty ;).